### ISSUE TITLE
Plotting 3D Contour Plot On Sundials Data Causes Exception

### ISSUE THREAD

**Issue by Matt-Cumber** [mantidproject/mantid#31256]
### Expected behavior
Produce an error warning or prevents the data from being loaded

### Actual behavior
Produces the error reporter

### Steps to reproduce the behavior
Load a data file from sundials data directory
Try to do a 3D contour plot 

### Platforms affected
all

**Comment by martyngigg** [comment]
@Matt-Cumber Could you clarify what the sundials data is and provide a link?

Is there any error message produced in the log that you can paste in to help whoever picks this up?

**Comment by Matt-Cumber** [comment]
Error message shown:
```
Traceback (most recent call last):
  File "/Applications/MantidWorkbenchNightly.app/Contents/MacOS/workbench/plugins/workspacewidget.py", line 184, in _do_plot_3D
    plot_function(workspaces)
  File "/Applications/MantidWorkbenchNightly.app/Contents/MacOS/mantid/plots/plotfunctions.py", line 58, in inner_func
    return func(workspaces, *args, **kwargs)
  File "/Applications/MantidWorkbenchNightly.app/Contents/MacOS/mantidqt/plotting/functions.py", line 353, in plot_contour
    ax.contour(ws, levels=DEFAULT_CONTOUR_LEVELS,
  File "/Applications/MantidWorkbenchNightly.app/Contents/MacOS/mantid/plots/mantidaxes.py", line 43, in wrapper
    func_value = func(self, *args, **kwargs)
  File "/Applications/MantidWorkbenchNightly.app/Contents/MacOS/mantid/plots/mantidaxes.py", line 997, in contour
    return self._plot_2d_func('contour', *args, **kwargs)
  File "/Applications/MantidWorkbenchNightly.app/Contents/MacOS/mantid/plots/mantidaxes.py", line 916, in _plot_2d_func
    plotfunctions_func(self, *args, **kwargs),
  File "/Applications/MantidWorkbenchNightly.app/Contents/MacOS/mantid/plots/axesfunctions.py", line 367, in contour
    return axes.contour(x, y, z, *args, **kwargs)
  File "/Applications/MantidWorkbenchNightly.app/Contents/MacOS/mantid/plots/mantidaxes.py", line 43, in wrapper
    func_value = func(self, *args, **kwargs)
  File "/Applications/MantidWorkbenchNightly.app/Contents/MacOS/mantid/plots/mantidaxes.py", line 997, in contour
    return self._plot_2d_func('contour', *args, **kwargs)
  File "/Applications/MantidWorkbenchNightly.app/Contents/MacOS/mantid/plots/mantidaxes.py", line 923, in _plot_2d_func
    return getattr(Axes, name)(self, *args, **kwargs)
  File "/Applications/MantidWorkbenchNightly.app/Contents/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/matplotlib/__init__.py", line 1599, in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
  File "/Applications/MantidWorkbenchNightly.app/Contents/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/matplotlib/axes/_axes.py", line 6422, in contour
    contours = mcontour.QuadContourSet(self, *args, **kwargs)
  File "/Applications/MantidWorkbenchNightly.app/Contents/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/matplotlib/contour.py", line 855, in __init__
    kwargs = self._process_args(*args, **kwargs)
  File "/Applications/MantidWorkbenchNightly.app/Contents/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/matplotlib/contour.py", line 1456, in _process_args
    x, y, z = self._contour_args(args, kwargs)
  File "/Applications/MantidWorkbenchNightly.app/Contents/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/matplotlib/contour.py", line 1514, in _contour_args
    x, y, z = self._check_xyz(args[:3], kwargs)
  File "/Applications/MantidWorkbenchNightly.app/Contents/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/matplotlib/contour.py", line 1551, in _check_xyz
    raise TypeError("Input z must be at least a 2x2 array.")
TypeError: Input z must be at least a 2x2 array.

Continue working.

```

I believe sundials is a directory used for elemental analysis. 

[Sundials.zip](https://github.com/mantidproject/mantid/files/6398020/Sundials.zip)

**Comment by AnthonyLim23** [comment]
ah, these data sets are just, `x, y, e` so you cant do a contour plot of the data

**Comment by AnthonyLim23** [comment]
I deleted the zip because it is not public data

**Comment by martyngigg** [comment]
I don't think this is important enough to do for 6.1?

**Comment by martyngigg** [comment]
It checks correctly anyway I guess it just needs a cleaner error capture.

**Comment by martyngigg** [comment]
Fixed by #31373
